### ISSUE TITLE
[order] pathGroup doesn't appear to work

### ISSUE THREAD

**Issue by nickzelei** [import-js/eslint-plugin-import#2008]
# Problem
I am configuring pathGroups, but they do not appear to be working.
No matter the group I set for the pattern, it always requires that it appear at the top of the external imports.
I've manually verified that `minimatch` is returning the correct output for some of the test paths that I'm working with.

# tsconfig.json
```
"compilerOptions": {
   ...
   "paths": {
     "@/*": ["./*"]
   }
}
```

# .eslintrc.js
```
  'import/order': [
    'error',
    {
      groups: ['builtin', 'external', ['sibling', 'parent'], 'index', 'internal', 'object'],
      pathGroups: [
        {
          pattern: '@/**',
          group: 'internal',
        },
      ],
      'newlines-between': 'always',
      alphabetize: {
        order: 'asc',
        caseInsensitive: true,
      },
    },
  ],
```

# Expected
```
import { StatusCodes } from 'http-status-codes';
import { Route, useRouter } from 'next-router-provider';
import React, { useCallback, useEffect, useRef, useState } from 'react';
import styled from 'styled-components';

import { useSession } from '@/client/providers/session';
import { pageRoutes } from '@/client/routes';
```

# Actual
```
import { useSession } from '@/client/providers/session';
import { pageRoutes } from '@/client/routes';
import { StatusCodes } from 'http-status-codes';
import { Route, useRouter } from 'next-router-provider';
import React, { useCallback, useEffect, useRef, useState } from 'react';
import styled from 'styled-components';
```

# versions
```
eslint 7.22.0
eslint-plugin-import 2.22.1
typescript 4.2.3
```
# minimatch outputs (both of which are correct)
```
> minimatch('@/client/types', '@/**')
true
> minimatch('@external-package/foo-package', '@/**')
false

```

**Comment by Jack-Barry** [comment]
It would be nice if this could be supported; Vue CLI projects use the `@/` prefix to point to source root by default and it would make sense for paths with a slash after the `@` to be differentiated between paths without a slash, `@/my-internal-module` vs.  `@external-module`

**Comment by nickzelei** [comment]
Ah yes, that seems to have done it @Jack-Barry - however that was not obvious! Is there any lingering effects of setting `pathGroupsExcludedImportTypes` to empty array? Like ideally it would be nice to get the defaults for that minus the alias check.

**Comment by Jack-Barry** [comment]
@nickzelei Indeed that does do it, and is cleaner - thanks for taking some time to look into it. I'm adding a second edit in my previous comment to point others to your answer.

**Comment by johnloven** [comment]
You can also use an eslint setting in order to interpret specific path patterns as internal:

```yaml
# .eslintrc.yml
settings:
  import/internal-regex: ^@/
```

https://github.com/benmosher/eslint-plugin-import/blob/master/README.md#importinternal-regex

**Comment by ljharb** [comment]
This seems resolved.

PRs to improve the docs are always welcome!
